packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -60,14 +43,7 @@ export interface TokenAuthorizerProps {
   readonly assumeRole?: iam.IRole;
 }
 
-/**
- * Token based lambda authorizer that recognizes the caller's identity as a bearer token,
- * such as a JSON Web Token (JWT) or an OAuth token.
- * Based on the token, authorization is performed by a lambda function.
- *
- * @resource AWS::ApiGateway::Authorizer
- */
-export class TokenAuthorizer extends Authorizer implements IAuthorizer {
+abstract class LambdaAuthorizer extends Authorizer implements IAuthorizer {
```

Comment:
Why?

Reply:
Not exporting this class indicates to the typescript compiler that type information need not be generated and exported.

If there is no type exported, when consumers of this library import `RequestAuthorizer` or `TokenAuthorizer` into their module, the typescript compiler would error saying that it cannot determine the entire type of this structure.

Reply:
That's not true, since the top-level subclass _is_ being exported. You can see this in effect because the unit/integ tests work correctly without issue.